src/olympia/addons/tests/test_models.py

```diff
@@ -1919,14 +1919,31 @@ class TestAddonDependencies(TestCase):
     def test_dependencies(self):
         ids = [3615, 3723, 4664, 6704]
         addon = Addon.objects.get(id=5299)
+        dependents = Addon.objects.in_bulk(ids)
 
-        for dependent_id in ids:
-            AddonDependency(
-                addon=addon,
-                dependent_addon=Addon.objects.get(id=dependent_id)).save()
+        for dependent in dependents.values():
+            AddonDependency(addon=addon, dependent_addon=dependent).save()
 
+        # Make sure all dependencies were saved correctly.
         assert sorted([a.id for a in addon.dependencies.all()]) == sorted(ids)
-        assert list(a.dependencies.all()) == a.all_dependencies
```

Comment:
That test was wrong, it should have used `addon` instead of `a`, and it would have failed because of the limit of `3` addons in dependencies (that exist on purpose, see 9be56d044207ab239c4a39a17062dc6f0208ef61)